public/blog/announcing-full-portable-solidity-debugger.md

```diff
@@ -0,0 +1,79 @@
+
+    image or gif will live here
+
+### What We've Built
+
+If you've followed Truffle development, you're familiar with the `truffle debug` command. This was the first incarnation of a "debugger" built into Truffle, though it wasn't much more than a code tracer. Our new debugger comes with everything you expect :
+
+* Code stepping (over, into, out, next, instruction, etc.)
+* Current code location, including the address of the running contract
+* Breakpoints
+* Watch expressions
+* Variable inspection (stack, memory _and_ storage)
+* Custom expression evaluation using Solidity variables
+
+On top of that, our new debugger is fully portable, built to integrate with any Javascript project or editor, like VSCode. Lastly, we built it to work with **any Ethereum client**, so you could debug transactions against the main Ethereum network if you so desired. 
+
+### How it Works
+
+To debug your Solidity code, you first need to make a transaction. Ethereum is a world computer with a history of all execution, so we can pull all the information we need from past transactions. After obtaining the transaction hash, simply run the following to kick off your debugging experience:
+
+```shell
+$ truffle debug <tx hash>
+```
+
+This will open up the debugging console which you can use to your heart's content! 
+
+    Insert another picture.
+
+The above user interface gives you access to the debugger features you need to step through your contracts' code, set breakpoints, evaluate expressions and inspect Solidity variables, providing you a better sense of what your code is doing. Under the hood, our debugger is interacting with your Ethereum client to gather all the data necessary to provide this information to you in a suitable manner. For developers reading this, the debugging library stores all this data in Redux, making the data easy to manage and making the library super portable. 
+
+Most of the work done by the debugger results in some type of mapping. The debugger gathers tons of disparate data, like the bytecode of your contracts, abstract representations of your code, instructions run during your transaction, etc., and combines them to produce information useful to the contract developer. Perhaps the debugger's most valuable property is variable inspection: Using all the data gathered, the debugger can determine which variables exist within your contract, which apply to each section of the code, and what their values are at any point in time. It can even show the values of complex data types, like structs, nested arrays and even mappings of structs. 
```

Comment:
Suggest bolding "variable inspection" since it's so key. Also, perhaps making that sentence a new paragraph.